src/utils/hooks/useClosestColorScheme.ts

```diff
@@ -0,0 +1,20 @@
+import { useMemo } from 'react';
+
+/**
+ * useClosestColorScheme - Returns 'light' / 'dark' based on the enforced color scheme closest to the element passed or `undefined` when no color scheme is enforced
+ */
+export const useClosestColorScheme = (element?: HTMLElement): 'light' | 'dark' | undefined =>
```

Comment:
Should we also subscribe for changes? In the case of the inverted colour mode, the class will be updated by the browser when OS changes the mode